bycycle/group/features.py

```diff
@@ -189,81 +269,90 @@ def compute_features_3d(sigs, fs, f_range, compute_features_kwargs=None,
+        df_features = [list(dfs) for dfs in zip(*df_features)]
 
-    if return_samples:
+    elif axis == (0, 1):
+        # Independently across the first two axes (i.e. for each signal)
+        sigs_2d = sigs.reshape(np.shape(sigs)[0]*np.shape(sigs)[1], np.shape(sigs)[2])
+        kwargs = kwargs[0] if len(kwargs) == 1 else kwargs
 
-        df_features, df_samples = df_features[0], df_features[1]
-        df_samples = _reshape_df(df_samples, sigs)
+        df_2d = compute_features_2d(sigs_2d, fs, f_range, compute_features_kwargs=kwargs,
+                                    return_samples=return_samples, n_jobs=n_jobs,
+                                    progress=progress, axis=0)
 
-    # Reshape returned features to match the first two dimensions of sigs
-    df_features = _reshape_df(df_features, sigs)
+    elif axis == None:
+        # Dependently across a flatten signal
+        sigs_2d = sigs.reshape(np.shape(sigs)[0]*np.shape(sigs)[1], np.shape(sigs)[2])
+        kwargs = kwargs[0] if len(kwargs) == 1 else kwargs
 
-    if return_samples:
-        return df_features, df_samples
+        df_2d = compute_features_2d(sigs_2d, fs, f_range, compute_features_kwargs=kwargs,
+                                    return_samples=return_samples, n_jobs=n_jobs,
+                                    progress=progress, axis=None)
 
-    return df_features
+    else:
 
+        raise ValueError("The axis kwarg must be either 0, 1, (0, 1) or None.")
 
-def _proxy(args, fs=None, f_range=None, return_samples=None):
-    """Proxy function to map kwargs and sigs together."""
+    if axis == (0, 1) or axis == None:
 
-    sig, kwargs = args[0], args[1:]
-    return compute_features(sig, fs=fs, f_range=f_range,
-                            return_samples=return_samples, **kwargs[0])
+         # Reshape
+         for dim0_idx in range(np.shape(sigs)[0]):
+            for dim1_idx in range(np.shape(sigs)[1]):
+                df_features[dim0_idx][dim1_idx] = df_2d[dim0_idx + dim1_idx]
 
+    return df_features
 
-def _reshape_df(df_features, sigs_3d):
-    """Reshape a list of dataframes."""
 
-    df_reshape = []
+def _proxy_2d(args, fs=None, f_range=None, return_samples=None):
+    """Proxy function to map kwargs and 2d sigs together."""
 
-    dim_b = np.shape(sigs_3d)[1]
+    sig, kwargs = args[0], args[1:]
+    return compute_features(sig, fs=fs, f_range=f_range,
+                            return_samples=return_samples, **kwargs[0])
 
-    max_idx = [i for i in range(dim_b, len(df_features)*dim_b, dim_b)]
-    min_idx = [i for i in range(0, len(df_features), dim_b)]
+def _proxy_3d(args, fs=None, f_range=None, return_samples=None):
+    """Proxy function to map kwargs and 3d sigs together."""
 
-    for lower, upper in zip(min_idx, max_idx):
-        df_reshape.append(df_features[lower:upper])
+    sigs, kwargs = args[0], args[1]
 
-    return df_reshape
+    return compute_features_2d(sigs, fs=fs, f_range=f_range, return_samples=return_samples,
```

Comment:
very much a nitpick, but for consistency, I would say order of arguments here should also match the function def.